pkg/destroy/azure/azure.go

```diff
@@ -80,19 +81,23 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 	}
 
 	return &ClusterUninstaller{
-		SubscriptionID:  session.Credentials.SubscriptionID,
-		TenantID:        session.Credentials.TenantID,
-		GraphAuthorizer: session.GraphAuthorizer,
-		Authorizer:      session.Authorizer,
-		InfraID:         metadata.InfraID,
-		Logger:          logger,
+		SubscriptionID:    session.Credentials.SubscriptionID,
+		TenantID:          session.Credentials.TenantID,
+		GraphAuthorizer:   session.GraphAuthorizer,
+		Authorizer:        session.Authorizer,
+		InfraID:           metadata.InfraID,
+		ResourceGroupName: metadata.Azure.ResourceGroupName,
+		Logger:            logger,
 	}, nil
 }
 
 // Run is the entrypoint to start the uninstall process.
 func (o *ClusterUninstaller) Run() error {
 	o.configureClients()
-	group := o.InfraID + "-rg"
+	group := o.ResourceGroupName
+	if len(group) == 0 {
+		group = o.InfraID + "-rg"
+	}
```

Comment:
If you move this into New statement above,  all initialization would be in one place. And function signature looks much nicer:
`deletePublicRecords(context.TODO(), o.zonesClient, o.recordsClient, o.privateZonesClient, o.privateRecordSetsClient, o.Logger, o.ResourceGroup)`

Reply:
fixed in 3932221